components/dm/command/scale_in.go

```diff
@@ -100,23 +102,46 @@ func ScaleInDMCluster(
 	}
 
 	// Cannot delete all DM DMMaster servers
-	if len(deletedDiff[dm.ComponentDMMaster]) == len(spec.Masters) {
+	if len(deletedDiff[dm.ComponentDMMaster]) == len(topo.Masters) {
 		return errors.New("cannot delete all dm-master servers")
 	}
 
+	maybeDestroyMonitor := func(instance spec.Instance) error {
```

Comment:
I think there is no need to destroy monitored instance in dm cluster since dm doesn't support monitored instance at all: https://github.com/pingcap/tiup/blob/master/components/dm/spec/topology_dm.go#L89

Reply:
Thanks for the advice, revert this commit